### PR TITLE
gi: support GVariant boolean -> Ruby

### PR DESCRIPTION
**Pull request by knatsakis**

### CODE REVIEW

**Comment by kou** [comment]
This pull request includes a merge commit. Please rebase on master and remove it.

File: gobject-introspection/ext/gobject-introspection/rb-gi-argument.c [hunk 1]
```diff
@@ -548,6 +548,10 @@ interface_variant_to_ruby(GVariant *variant)
         const char *value;
         g_variant_get(variant, "s", &value);
         rb_value = CSTR2RVAL(value);
+    } else if (g_variant_type_equal(type, G_VARIANT_TYPE_BOOLEAN)) {
+        const gboolean *value;
+	g_variant_get(variant, "b", &value);
```
Comment (kou) [inline gobject-introspection/ext/gobject-introspection/rb-gi-argument.c]: A tab is used.

File: gobject-introspection/ext/gobject-introspection/rb-gi-argument.c [hunk 2]
```diff
@@ -548,6 +548,10 @@ interface_variant_to_ruby(GVariant *variant)
...
+	rb_value = CBOOL2RVAL(value);
```
Comment (kou) [inline gobject-introspection/ext/gobject-introspection/rb-gi-argument.c]: `*value`?

**Comment by kou** [comment]
Could you provide a test case or description how to test the change by us?

File: gobject-introspection/ext/gobject-introspection/rb-gi-argument.c (hunk 1 above)
Comment (knatsakis) [inline gobject-introspection/ext/gobject-introspection/rb-gi-argument.c]: Fixed

File: gobject-introspection/ext/gobject-introspection/rb-gi-argument.c (hunk 2 above)
Comment (knatsakis) [inline gobject-introspection/ext/gobject-introspection/rb-gi-argument.c]: Fixed. I didn't do appropriate testing it seems.

**Comment by knatsakis** [comment]
I have removed the merge commit, replaced tabs with spaces for indentation and fixed the wrong use of *value.

I am testing this with Gio::Settings.
You can find a boolean gsettings value with _dconf-editor_ and try to retrieve the value. E.g.:

From a shell:
`gsettings get org.mate.background show-desktop-icons`
`false`
`gsettings get org.mate.background background-fade`
`true`

From ruby:

`[1] pry(main)> require 'gio2'`
`=> true`
`[2] pry(main)> Gio::Settings::new('org.mate.background').get_value('show-desktop-icons')`
`=> false`
`[3] pry(main)> Gio::Settings::new('org.mate.background').get_value('background-fade')`
`=> true`

**Comment by kou** [comment]
Thanks.
I've implemented it in different way (we already have implementation!) and added tests for it by `Gio::Settings`.

**Pull request closed without merging**